core/lib/object_store/src/objects.rs

```diff
@@ -63,6 +65,30 @@ macro_rules! serialize_using_bincode {
     };
 }
 
+/// Derives [`StoredObject::serialize()`] and [`StoredObject::deserialize()`] using
+/// the `json` (de)serializer. Should be used in `impl StoredObject` blocks.
+
+impl StoredObject for StorageLogsSnapshot {
+    const BUCKET: Bucket = Bucket::StorageSnapshot;
+    type Key<'a> = StorageLogsSnapshotKey;
+
+    fn encode_key(key: Self::Key<'_>) -> String {
+        format!(
+            "storage_logs_{}_part{}.json",
+            key.l1_batch_number, key.chunk_id
+        )
+    }
+
+    //TODO use better language agnostic serialization format like protobuf
```

Comment:
Intuitively, using JSON is a non-starter because of chunk size (order of 1 GiB); a binary format would be more compact and could support streaming deserialization etc. (esp. if it's Protobuf). We'll likely require some compression as well (although it may be sufficient to enable on the HTTP level).

Reply:
I have used HTTP compression a lot and I'm a big fan of it, but here I am a bit concerned about storage size on GS as a single uncompressed snapshot would be more than 100GB and I would expect them to compress really well.

Reply:
Protobuf seems fine for consistency, but I'm not an expert w.r.t. best practices; probably we need to discuss how to do it properly with folks who have enough experience here.

Reply:
> a single uncompressed snapshot would be more than 100GB

you mean one chunk??

Reply:
We'll be compressing chunks individually won't we

Reply:
I mean 100GB for a whole snapshot, but it will quickly add up to terabytes if we are to create them regularly. I do have quite a bit of experience with using them from my previous company, we used them on almost all our APIs that required high bandwith

Reply:
yes, I expected compression to be on the individual file level